### PR TITLE
Fixes for errors reported by cppcheck

### PR DESCRIPTION
**Pull request by mc-hamster**
Fix https://github.com/siara-cc/Unishox2/issues/39

This was discovered while integrating this compression library with the Meshtastic project.

Leaving this as a draft until I complete some internal testing.

### CODE REVIEW

File: unishox2.c [hunk 1]
```diff
@@ -129,11 +129,9 @@ void init_coder() {
   for (int i = 0; i < 3; i++) {
     for (int j = 0; j < 28; j++) {
       byte c = usx_sets[i][j];
-      if (c != 0 && c > 32) {
```
Comment (siara-cc) [inline unishox2.c:132]: The `c > 32` is required.  `c != 0` is redundant, but it indicates 0 values are not to be processed.

File: unishox2.c [hunk 2]
```diff
@@ -147,32 +145,31 @@ unsigned int usx_mask[] = {0x80, 0xC0, 0xE0, 0xF0, 0xF8, 0xFC, 0xFE, 0xFF};
...
 
-  byte cur_bit;
-  byte blen;
-  unsigned char a_byte;
-  int oidx;
 
   //printf("%d,%x,%d,%d\n", ol, code, clen, state);
 
   while (clen > 0) {
-     cur_bit = ol % 8;
```
Comment (siara-cc) [inline unishox2.c:158]: Is it possible to restore the indent?  It will help see the changes clearly.

**Review by siara-cc — CHANGES_REQUESTED** [review]

File: unishox2.c (hunk 1 above)
Comment (mc-hamster) [reply unishox2.c:132]: Fixed

File: unishox2.c (hunk 2 above)
Comment (mc-hamster) [reply unishox2.c:158]: Fixed here.

**Comment by siara-cc** [comment]
@mc-hamster  Thank you for the PR. I am merging it.

**Pull request merged**